python/plugins/processing/algs/qgis/CreateConstantRaster.py

```diff
@@ -61,28 +76,35 @@ def displayName(self):
         return self.tr('Create constant raster layer')
 
     def processAlgorithm(self, parameters, context, feedback):
-        layer = self.parameterAsRasterLayer(parameters, self.INPUT, context)
+        extent = self.parameterAsExtent(parameters, self.EXTENT, context)
+        crs = self.parameterAsCrs(parameters, self.TARGET_CRS, context)
         value = self.parameterAsDouble(parameters, self.NUMBER, context)
+        pixelSize = self.parameterAsDouble(parameters, self.PIXEL_SIZE, context)
 
         outputFile = self.parameterAsOutputLayer(parameters, self.OUTPUT, context)
+        outputFormat = raster.formatShortNameFromFileName(outputFile)
```

Comment:
I agree this should be a generic method in core, as it useful for other plugins. I will look into it